### PR TITLE
GEODE-10286: handle CancelException in PersistenceAdvisor.close

### PR DESCRIPTION
**Pull request by jinmeiliao**
* GEODE-10287: do distAdvisor.close after waitForCurrentOperations()

### CODE REVIEW

File: geode-core/src/main/java/org/apache/geode/internal/cache/persistence/PersistenceAdvisorImpl.java [hunk 1]
```diff
@@ -1161,11 +1162,19 @@ public void checkInterruptedByShutdownAll() {
...
+      synchronized (this) {
+        isClosed = true;
+        persistentMemberManager.removeRevocationListener(profileChangeListener);
+        cacheDistributionAdvisor.removeProfileChangeListener(profileChangeListener);
+        persistentStateListeners = Collections.emptySet();
+        releaseTieLock();
+      }
+    } catch (CancelException e) {
+      logger.debug("persistence advisor close abridged due to shutdown", e);
+    } catch (Exception ex) {
```
Comment (jchen21) [inline geode-core/src/main/java/org/apache/geode/internal/cache/persistence/PersistenceAdvisorImpl.java:1175]: If there is Exception other than `CancelException`, I don't think the Exception can be ignored. It should be thrown.

File: geode-core/src/main/java/org/apache/geode/internal/cache/DistributedRegion.java [hunk 2]
```diff
@@ -2605,6 +2604,7 @@ protected void distributedRegionCleanup(RegionEventImpl event) {
       }
     }
     waitForCurrentOperations();
+    distAdvisor.close();
```
Comment (DonalEvans) [inline geode-core/src/main/java/org/apache/geode/internal/cache/DistributedRegion.java:2607]: It seems that this change is leading to the `ConnectionCloseSSLTLSDUnitTest.connectionWithHungReaderIsCloseableAndUnhangsReader()` test hanging

File: geode-core/src/main/java/org/apache/geode/internal/cache/persistence/PersistenceAdvisorImpl.java (hunk 1 above)
Comment (jinmeiliao) [reply geode-core/src/main/java/org/apache/geode/internal/cache/persistence/PersistenceAdvisorImpl.java:1175]: as we are in the cleanup phase, if we throw the exception, the rest of the cleanup will be skipped, which is not what we want. This behavior is consistent with other advisor close method in the `DistributedRegion`, as in DLockService.close and `DistributionAdvisor.close`

File: geode-core/src/main/java/org/apache/geode/internal/cache/DistributedRegion.java (hunk 2 above)
Comment (jinmeiliao) [reply geode-core/src/main/java/org/apache/geode/internal/cache/DistributedRegion.java]: Yeah, looks like this is the reason, I am going to leave GEODE-10287 out of this PR.

**Review by DonalEvans — APPROVED** [review]

**Review by gesterzhou — APPROVED** [review]

**Review by dschneider-pivotal — APPROVED** [review]

**Pull request merged**